module_auto_update/views/module_views.xml

```diff
@@ -0,0 +1,77 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <!-- Module Search View -->
+    <record id="module_view_search" model="ir.ui.view">
+        <field name="name">updates.module.search</field>
+        <field name="model">ir.module.module</field>
+        <field name="inherit_id" ref="base.view_module_filter"/>
+        <field name="arch" type="xml">
+            <field name="category_id" position="after">
+                <filter name="upgrade_available" string="Upgrade Available" domain="[('upgrade_available', '=', True)]"/>
+                <field name="upgrade_available"/>
+            </field>
+        </field>
+    </record>
+
+    <!-- Module Tree View -->
+    <record id="module_view_tree" model="ir.ui.view">
+        <field name="name">updates.module.tree</field>
+        <field name="model">ir.module.module</field>
+        <field name="inherit_id" ref="base.module_tree"/>
+        <field name="priority" eval="20"/>
+        <field name="arch" type="xml">
+            <field name="state" position="after">
+                <field name="upgrade_available"/>
+            </field>
+        </field>
+    </record>
+
+    <!-- Module Form View -->
+    <record id="update_module_form" model="ir.ui.view">
+        <field name="name">ir.module.module.form</field>
+        <field name="model">ir.module.module</field>
+        <field name="inherit_id" ref="base.module_form"/>
+        <field name="arch" type="xml">
+            <field name="installed_version" position="after">
+                <field name="checksum_installed"/>
+                <field name="checksum_dir"/>
+                <field name="upgrade_available"/>
+            </field>
+        </field>
+    </record>
+
+    <!--Open Updates Action (updates apps list first)-->
+    <record id="module_action_open_updates" model="ir.actions.server">
+        <field name="name">Open Updates and Update Apps List Server Action</field>
+        <field name="model_id" ref="model_ir_module_module"/>
+        <field name="code">
+            if model.update_list():
+                action = {
+                    'name': 'Updates',
+                    'type': 'ir.actions.act_window',
+                    'res_model': 'ir.module.module',
+                    'view_type': 'form',
+                    'view_mode': 'tree,form',
+                    'target': 'main',
+                    'context': '{"search_default_upgrade_available": 1}',
+                }
```

Comment:
If we use the state as said above, could we move all the checksum computing into `update_list()`, so we don't have to create additional actions, menu items and methods? This way, if the admin uses the standard system to upgrade the addons list, he will always update their checksums too. It seems like what he'd expect to happen.